### PR TITLE
[enhance](mtmv)Change the way to verify the existence of partition names when refreshing MTMV

### PR DESCRIPTION
**Pull request by zddr**
### What problem does this PR solve?

Previously, when refreshing the materialized view according to the partition name, the existing partition of the materialized view was used to verify whether the partition name existed
After the change, the partition that should be present after refreshing the materialized view is used for verification

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:
Change the way to verify the existence of partition names when refreshing MTMV
### Release note

Change the way to verify the existence of partition names when refreshing MTMV

### Check List (For Author)

- Test 
    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [ ] No.
    - [x] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by zddr** [comment]
run p0

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/info/RefreshMTMVInfo.java [hunk 1]
```diff
@@ -67,13 +73,41 @@ public void analyze(ConnectContext ctx) {
...
         } catch (org.apache.doris.common.AnalysisException | MetaNotFoundException | DdlException e) {
             throw new AnalysisException(e.getMessage());
         }
     }
 
+    private void checkPartitionExist(MTMV mtmv) throws org.apache.doris.common.AnalysisException {
+        MTMVRelatedTableIf relatedTable = mtmv.getMvPartitionInfo().getRelatedTable();
+        // todo: Unified locking sequence
+        mtmv.readLock();
+        relatedTable.readLock();
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/info/RefreshMTMVInfo.java:87]: yes should lock by id, maybe u could use
```
List<TableIf> tables = Lists.newArrayList(mtmv, relatedTable);
tables.sort(Comparator.comparing(TableIf::getId));
MetaLockUtils.readLockTables(tables);
try {
    ...
} finally {
    MetaLockUtils.readUnlockTables(tableIfs);
}
```

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run p0

**Review by seawinde — APPROVED** [review]

**Review by morrySnow — APPROVED** [review]

**Pull request merged**